test/intergration_test.go

```diff
@@ -0,0 +1,239 @@
+	}
+	web.Config.Base = "/dbs"
+	web.Config.DBFile = dbfile
+	web.Config.LexiconFile = lexiconFile
+	web.Config.ServerCrt = ""
+	web.Config.ServerKey = ""
+	web.Config.ServerType = serverType
+	web.Config.LogFile = "/tmp/dbs2go-test.log"
+	web.Config.Verbose = 0
+
+	initTestLimiter(t, "100-S")
+
+	ts := httptest.NewServer(web.Handlers())
+
+	return ts
+}
+
+// injects dbs records
+func injectDBRecord(t *testing.T, rec dbs.DBRecord, method string, url string, endpoint string, handler func(http.ResponseWriter, *http.Request)) []dbs.Record {
+	data, err := json.Marshal(rec)
+	if err != nil {
+		t.Fatal(err.Error())
+	}
+	reader := bytes.NewReader(data)
+	req := newreq(t, method, url, endpoint, reader, nil)
+	req.Header.Add("Accept", "application/json")
+	req.Header.Set("Content-Type", "application/json")
+	r, err := http.DefaultClient.Do(req)
+	if err != nil {
+		t.Fatal(err)
+	}
+	defer r.Body.Close()
+
+	if r.StatusCode != http.StatusOK {
+		t.Fatal(err)
+	}
+
+	rr, err := respRecorder("GET", endpoint, nil, handler)
+	if err != nil {
+		t.Error(err)
+	}
+
+	var records []dbs.Record
+	data = rr.Body.Bytes()
+	err = json.Unmarshal(data, &records)
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	return records
+}
+
+// fetches data from url and endpoint
+func getData(t *testing.T, url string, endpoint string, params url.Values) ([]dbs.Record, int) {
+	r, err := http.DefaultClient.Do(newreq(t, "GET", url, endpoint, nil, params))
+	if err != nil {
+		t.Fatal(err)
+	}
+	defer r.Body.Close()
+
+	if r.StatusCode != http.StatusOK {
+		t.Fatalf("Bad status code: %v", r.StatusCode)
+	}
+
+	var d []dbs.Record
+	err = json.NewDecoder(r.Body).Decode(&d)
+	if err != nil {
+		t.Fatalf("Failed to decode body, %v", err)
+	}
+
+	return d, r.StatusCode
+}
+
+// creates an http request for testing
+func newreq(t *testing.T, method string, hostname string, endpoint string, body io.Reader, params url.Values) *http.Request {
+	url2, err := url.Parse(hostname)
+	if err != nil {
+		t.Fatal(err)
+	}
+	url2.Path = endpoint
+	url2.RawQuery = params.Encode()
+
+	r, err := http.NewRequest(method, url2.String(), body)
+	if err != nil {
+		t.Fatal(err)
+	}
+	return r
+}
+
+// compares received response to expected
+func verifyResponse(t *testing.T, received []dbs.Record, expected map[string]interface{}, fields []string) {
+	for _, r := range received {
+		for _, f := range fields {
+			if r[f] != expected[f] {
+				if strings.Contains(f, "id") || f == "creation_date" {
+					if r[f] == nil {
+						t.Fatalf("ID field empty")
+					}
+				} else {
+					t.Fatalf("Incorrect %s: Expected %v, Received: %v", f, expected[f], r[f])
+				}
+			}
+		}
+	}
+}
+
+// run test workflow for a single endpoint
+func runTestWorkflow(t *testing.T, tsR *httptest.Server, tsW *httptest.Server, endpoint string, hdlr func(http.ResponseWriter, *http.Request), dbrec dbs.DBRecord, params url.Values, fields []string) {
+	emap := remapRecord(t, dbrec)
+
+	t.Run("Test empty GET", func(t *testing.T) {
```

Comment:
For DBS APIs the request to end-point without parameters is not allowed. For instance, the FILES tables is very big and therefore call to `/files` API is prohibited to avoid fetching result of entire table (in production it has O(100M) rows). As such, I suggest to add additional flag to `runTestWorkflow` which will trigger this test.

Reply:
no, only to some APIs. For instance, `datatiers`, `datasetaccesstypes`, `physicsgroups` and even `datasets` API may not require input parameters, but `blocks`, `flles` do require input parameters. It is use-case driven and based on amount of information stored in DBS tables. FYI:
- small tables tiers, dataset_access_types,  physics_groups
- medium tables with O(100K-1M) rows is datasets
- large tables with O(10M) rows is blocks
- very large tables with O(100M) rows is files

For large tables we do not allow to fetch all rows in API due to amount of stored data in those tables. Please fetch this stats: https://cmsweb.cern.ch/dbs/prod/global/DBSReader/dbstats which will tell you table size and other stats. The new server provides it now.